planner/cascades/integration_test.go

```diff
@@ -292,3 +292,35 @@ func (s *testIntegrationSuite) TestTopN(c *C) {
 		tk.MustQuery(sql).Check(testkit.Rows(output[i].Result...))
 	}
 }
+
+func (s *testIntegrationSuite) TestCascadePlannerHashedPartTable(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists pt1")
+	tk.MustExec("create table pt1(a bigint, b bigint) partition by hash(a) partitions 4")
+	tk.MustExec("set @@tidb_enable_cascades_planner = 1")
+
+	tk.MustExec(`insert into pt1 values(1,10)`)
+	tk.MustExec(`insert into pt1 values(2,20)`)
+	tk.MustExec(`insert into pt1 values(3,30)`)
+	tk.MustExec(`insert into pt1 values(4,40)`)
+	tk.MustExec(`insert into pt1 values(5,50)`)
+
+	tk.MustQuery(`desc select * from pt1`).Check(testkit.Rows(
```

Comment:
Could you move this test case to the json file like other test functions in this file? So we can use a flag `-record` to generate the test result, which would be more friendly for the future changes.